provider/core/src/datagen/payload.rs

```diff
@@ -53,7 +53,7 @@ impl core::fmt::Debug for ExportBox {
 impl<M> UpcastDataPayload<M> for ExportMarker
 where
     M: DataMarker,
-    M::Yokeable: Sync,
+    M::Yokeable: Sync + Send,
```

Comment:
This is technically breaking, but I think it's almost impossible to have `!Send` data structs, as they'd have to contain raw pointers or `Rc`. If we don't want this we can remove locale-level parallelism and generate all locales for one key on the same thread.

Reply:
I lean toward shipping this and rolling back if anyone has problems